components/authn-service/tokens/pg/migrate.go

```diff
@@ -1,183 +1,76 @@
 package pg
 
 import (
+	"context"
 	"database/sql"
-	"fmt"
 
+	"github.com/mattes/migrate"
+	_ "github.com/mattes/migrate/database/postgres" // make database available
+	_ "github.com/mattes/migrate/source/file"       // make source available
 	"github.com/pkg/errors"
-
-	"github.com/chef/automate/components/authn-service/constants"
 )
 
-// migrate tries to execute all the migrations we know of
-func migrate(c *sql.DB) error {
-	// Acquire advisory lock
-	_, err := c.Exec("SELECT pg_advisory_lock($1);", constants.AdvisoryLockID)
+// runMigrations tries to execute all the migrations we know of
+func runMigrations(db *sql.DB, url, path string) error {
+	// We're migrating to golang-migrate/migrate instead of this hand-rolled code:
+	// If it exists, we're migrating the existing migration table a golang-migrate conform
+	// schema_migrations table.
+	// When this is done, we drop that table, so the code here will only run once.
+	// We then run the usual "up" migration code, which will pick up where we've left off.
+	// Existing migrations have been converted into up.sql files accordingly, keeping
+	// the numbering intact.
+
+	ctx := context.TODO()
+	tx, err := db.BeginTx(ctx, nil)
 	if err != nil {
-		return errors.Wrapf(err, "acquiring advisory lock %d", constants.AdvisoryLockID)
+		return errors.Wrap(err, "being transaction")
```

Comment:
typo here: being -> beginning

Reply:
I've wanted to say "begin", but "beginning" works for me, too. 😃